CRM/Contact/Import/Parser/Contact.php

```diff
@@ -229,10 +235,7 @@ public function init() {
     }
 
     $this->_updateWithId = FALSE;
-    if (in_array('id', $this->_mapperKeys) || ($this->_externalIdentifierIndex >= 0 && in_array($this->_onDuplicate, array(
```

Comment:
In fact this change here is the only functional change. The rest is all tidy up & test support